docs/appendix-A.rst

```diff
@@ -0,0 +1,1783 @@
+   :language: fortran
+   :start-after: start no_return_no_arguments
+   :end-before: end no_return_no_arguments
+   :dedent: 8
+
+If wrapping a C++ library, a function with a C API will be created
+that Fortran can call.
+
+.. literalinclude:: ../regression/reference/tutorial/wrapTutorial.cpp
+   :language: c
+   :start-after: start TUT_no_return_no_arguments
+   :end-before: end TUT_no_return_no_arguments
+
+Fortran usage:
+
+.. code-block:: fortran
+
+    use tutorial_mod
+    call no_return_no_arguments
+
+The C++ usage is similar:
+
+.. code-block:: c++
+
+    #include "tutorial.hpp"
+
+    using namespace tutorial;
+    NoReturnNoArguments();
+
+
+Numeric Types
+-------------
+
+.. ############################################################
+
+.. _example_PassByValue:
+
+PassByValue
+^^^^^^^^^^^
+
+C library function:
+
+.. literalinclude:: ../regression/run/clibrary/clibrary.c
+   :language: c
+   :start-after: start PassByValue
+   :end-before: end PassByValue
+
+YAML:
+
+.. code-block:: yaml
+
+    - decl: double PassByValue(double arg1, int arg2)
+
+Both types are supported directly by the ``iso_c_binding`` module
+so there is no need for a Fortran function.
+The C function can be called directly by the Fortran interface
+using the ``bind(C)`` keyword.
+
+Calls C via the interface:
+
+.. literalinclude:: ../regression/reference/clibrary/wrapfclibrary.f
+   :language: fortran
+   :start-after: start pass_by_value
+   :end-before: end pass_by_value
+   :dedent: 8
+
+Fortran usage:
+
+.. code-block:: fortran
+
+    real(C_DOUBLE) :: rv_double
+    rv_double = pass_by_value(1.d0, 4)
+    call assert_true(rv_double == 5.d0)
+
+.. ############################################################
+
+.. _example_PassByReference:
+
+PassByReference
+^^^^^^^^^^^^^^^
+
+C library function:
+
+.. literalinclude:: ../regression/run/clibrary/clibrary.c
+   :language: c
+   :start-after: start PassByReference
+   :end-before: end PassByReference
+
+YAML:
+
+.. code-block:: yaml
+
+    - decl: void PassByReference(double *arg1+intent(in), int *arg2+intent(out))
+
+Calls C via the interface:
+
+.. literalinclude:: ../regression/reference/clibrary/wrapfclibrary.f
+   :language: fortran
+   :start-after: start pass_by_reference
+   :end-before: end pass_by_reference
+   :dedent: 8
+
+Example usage:
+
+.. code-block:: fortran
+
+    integer(C_INT) var
+    call pass_by_reference(3.14d0, var)
+    call assert_equals(3, var)
+
+.. ############################################################
+
+.. _example_Sum:
+
+Sum
+^^^
+
+C++ library function:
+
+.. literalinclude:: ../regression/run/pointers/pointers.cpp
+   :language: c
+   :start-after: start Sum
+   :end-before: end Sum
+
+:file:`pointers.yaml`:
+
+.. code-block:: yaml
+
+   - decl: void Sum(int len +implied(size(values)),
+                    int *values +dimension(:)+intent(in),
+                    int *result +intent(out))
+
+:file:`wrappointers.cpp`:
```

Comment:
It might be a little confusing here that the function is prefixed by `POI`.

Perhaps it would help to mention (here or at the top of the file) that some of the generated functions are in different libraries, which provide different library prefixes?

```cpp
void POI_sum(int len, int * values, int * result)
{
// splicer begin function.sum
    Sum(len, values, result);
    return;
// splicer end function.sum
}
```

Reply:
I added a comment in this particular section since was the first one with a C wrapper.  It mentions C_prefix.  I didn't duplicate it for every example.